### ISSUE TITLE
Server crash related to attributes class/restorative potion

### ISSUE THREAD

**Issue by ShibeTemple** [XzeroAir/Trinkets-1.12.2#44]
Hello! Your mod is awesome, however, I ran into this fatal error on my dedicated server. I have attached the crash log!

```
java.lang.NoSuchMethodError: net.minecraft.entity.ai.attributes.IAttributeInstance.func_142049_d()V
    at xzeroair.trinkets.attributes.RaceAttribute.RaceAttribute.removeAllModifiers(RaceAttribute.java:124)
    at xzeroair.trinkets.items.potions.PotionRestorative.removeAttributesModifiersFromEntity(PotionRestorative.java:70)
    at net.minecraft.entity.EntityLivingBase.onFinishedPotionEffect(EntityLivingBase.java:829)
    at net.minecraft.entity.EntityLivingBase.updatePotionEffects(EntityLivingBase.java:611)
    at net.minecraft.entity.EntityLivingBase.onEntityUpdate(EntityLivingBase.java:353)
    at net.minecraft.entity.EntityLiving.onEntityUpdate(EntityLiving.java:208)
    at net.minecraft.entity.Entity.onUpdate(Entity.java:389)
    at net.minecraft.entity.EntityLivingBase.onUpdate(EntityLivingBase.java:2096)
    at net.minecraft.entity.EntityLiving.onUpdate(EntityLiving.java:295)
    at net.minecraft.entity.monster.EntityMob.onUpdate(EntityMob.java:50)
    at org.spongepowered.common.event.tracking.TrackingUtil.tickEntity(TrackingUtil.java:167)
    at net.minecraft.world.WorldServer.redirect$zlm000$onCallEntityUpdate(WorldServer.java:4761)
    at net.minecraft.world.World.updateEntityWithOptionalForce(World.java:1996)
    at net.minecraft.world.WorldServer.updateEntityWithOptionalForce(WorldServer.java:832)
    at net.minecraft.world.World.updateEntity(World.java:1958)
    at net.minecraft.world.World.updateEntities(World.java:1762)
    at net.minecraft.world.WorldServer.updateEntities(WorldServer.java:3931)
    at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:767)
    at net.minecraft.server.dedicated.DedicatedServer.updateTimeLightAndEntities(DedicatedServer.java:397)
    at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:668)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
    at java.lang.Thread.run(Unknown Source)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Entity being ticked --
  Entity Type: minecraft:zombie (net.minecraft.entity.monster.EntityZombie)
  Entity ID: 1502958
  Entity Name: Zombie
  Entity's Exact location: -5095.70, 18.40, -3128.30
  Entity's Block location: World: (-5096,18,-3129), Chunk: (at 8,1,7 in -319,-196; contains blocks -5104,0,-3136 to -5089,255,-3121), Region: (-10,-7; contains chunks -320,-224 to -289,-193, blocks -5120,0,-3584 to -4609,255,-3073)
  Entity's Momentum: 0.00, 0.31, 0.00
  Entity's Passengers: []
  Entity's Vehicle: ~~ERROR~~ NullPointerException: null
  Entity NBT: {HurtByTimestamp:129,ForgeData:{SpongeData:{CanGrief:1b},"spawnercontrol:spawnerPos":{pos:-1400226782907452L,dimension:0},"Potion Core - Health Fix":0.0f,"Potion Core - Jump Boost Updated":1b},RelativeAABB:[-0.30000001192092896d,0.0d,-0.30000001192092896d,0.30000001192092896d,1.9500000476837158d,0.30000001192092896d],Attributes:[{Base:-2.0d,Modifiers:[{UUIDMost:2189974539282499591L,UUIDLeast:-6477178668949725305L,Amount:1.0d,Operation:0,Name:"normal"}],Name:"tc.mobmod"},{Base:0.0d,Name:"tc.mobmodtaint"},{Base:1.0d,Name:"potioncore.projectileDamage"},{Base:1.0d,Name:"potioncore.magicDamage"},{Base:1.0d,Name:"potioncore.jumpHeight"},{Base:1.0d,Name:"potioncore.damageResistance"},{Base:0.0d,Name:"potioncore.magicShielding"},{Base:1.0d,Name:"xat.entityRace"},{Base:20.0d,Name:"generic.maxHealth"},{Base:0.0d,Modifiers:[{UUIDMost:2468065482386326952L,UUIDLeast:-6471885000904874503L,Amount:0.007454570365389999d,Operation:0,Name:"Random spawn bonus"}],Name:"generic.knockbackResistance"},{Base:0.23000000417232513d,Name:"generic.movementSpeed"},{Base:2.0d,Name:"generic.armor"},{Base:0.0d,Name:"generic.armorToughness"},{Base:1.0d,Name:"forge.swimSpeed"},{Base:35.0d,Modifiers:[{UUIDMost:-3812564079743317690L,UUIDLeast:-8993623328903430637L,Amount:-0.07404417786360155d,Operation:1,Name:"Random spawn bonus"}],Name:"generic.followRange"},{Base:3.0d,Name:"generic.attackDamage"},{Base:0.0d,Name:"zombie.spawnReinforcements"}],Invulnerable:0b,FallFlying:0b,PortalCooldown:0,AbsorptionAmount:0.0f,FallDistance:0.0f,DeathTime:2s,ForgeCaps:{"tinkertoolleveling:entityxp":[],"dsurround:data":{a:0b,f:1b},"xat:race":{default_height:0.0f,default_width:0.0f,race:"Human",bonus_mana:0,prev_race:"Human",login:1b,first_login:0b,target:100,size:100,mana:100.0f,width:0.0f,transformed:0b,fake:0b,height:0.0f,affinity:1.0f},"llibrary:extendedentitydatacapability":{"mm:living":{frozenYaw:0.0f,freezeProgress:0.0f,frozenPitch:0.0f,prevHasAI:0b,freezeDecayDelay:0,frozenYawHead:0.0f,frozenLimbSwingAmount:0.0f,frozenRenderYawOffset:0.0f,frozenSwingProgress:0.0f},"Ice And Fire - Siren Property Tracker":{CharmedBySiren:0b,SirenID:0},"Ice And Fire - Frozen Property Tracker":{TicksUntilUnfrozen:0,IsFrozen:0b},"Ice and Fire - Player Property Tracker":{DismountedDragon:0b,GauntletDamage:0,DreadPortalZ:0,DreadPortalY:0,DreadPortalX:0},"Ice And Fire - Chain Property Tracker":{ConnectedEntities:[]},"Ice And Fire - Stone Property Tracker":{StoneBreakLvl:0,TurnedToStone:0b}},"mujmajnkraftsbettersurvival:nunchakucombo":{Time:0,Power:0.0f},"mujmajnkraftsbettersurvival:weaponeff":{Hits:0},"twilightforest:cap_shield":{permshields:0,tempshields:0},"mujmajnkraftsbettersurvival:spearsin":[],"somanyenchantments:player_capabilities":{isResurrecting:0b,moralitasDamage:0.0d}},HandDropChances:[0.085f,0.085f],PersistenceRequired:0b,Motion:[0.0d,0.3136000119447709d,0.0d],Leashed:0b,UUIDLeast:-8630132620515304462L,Health:0.0f,LeftHanded:0b,Air:300s,OnGround:0b,Dimension:0,Rotation:[286.9326f,0.0f],UpdateBlocked:0b,HandItems:[{},{}],ArmorDropChances:[0.085f,0.085f,0.085f,0.085f],UUIDMost:-4126734658442081326L,Pos:[-5095.699999988079d,18.400000005960464d,-3128.300000011921d],CanBreakDoors:0b,Fire:98s,ArmorItems:[{},{},{},{}],CanPickUpLoot:0b,HurtTime:8s,SurgeAABB:[-5096.0d,18.400000005960464d,-3128.600000023842d,-5095.399999976158d,20.35000005364418d,-3128.0d],ActiveEffects:[{Ambient:0b,CurativeItems:[{id:"minecraft:milk_bucket",Count:1b,Damage:0s}],ShowParticles:1b,Duration:166,Id:104,Amplifier:0b}]}

-- Affected level --
  Level name: world
  All players: 5 total; [GCEntityPlayerMP['Noitnettaa'/1325048, l='world', x=-5094.56, y=18.00, z=-3131.99], GCEntityPlayerMP['Skullpoptarts'/1369427, l='world', x=123.13, y=32.00, z=-334.90], GCEntityPlayerMP['Ant_on'/1406147, l='world', x=-3053.98, y=51.00, z=-1341.53], GCEntityPlayerMP['Grax9'/1481228, l='world', x=119.62, y=1.00, z=-377.57], GCEntityPlayerMP['ReligionDead21'/1415401, l='world', x=122.88, y=1.00, z=-375.94]]
  Chunk stats: ServerChunkCache: 2008 Drop: 0
  Level seed: -2205359657533471418
  Level generator: ID 12 - RTG, ver 0. Features enabled: true
  Level generator options: 
  Level spawn location: World: (132,64,252), Chunk: (at 4,4,12 in 8,15; contains blocks 128,0,240 to 143,255,255), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
  Level time: 16144711 game time, 6224689 day time
  Level dimension: 0
  Level storage version: 0x04ABD - Anvil
  Level weather: Rain time: 71221 (now: false), thunder time: 48159 (now: false)
  Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
```

Let me know if you need any additional information. I will try my best to reproduce the issue, however, I feel as if this is unlikely.

**Comment by XzeroAir** [comment]
Hello, and omg... apparently there is something wrong with my notifications for github, because I only saw this by checking the website manually, =_=
Glad you Like the mod, I'm really sorry it took so long to reply.
@ShibeTemple can you provide which version of T&B you're using

**Comment by XzeroAir** [comment]
I think I figured out the issue, it seems in 0.30, I mistakenly used a super method in the code for the Restorative potion, which is fine for players most of the time, but it breaks Zombies/Other mobs for some reason, I've actually heard someone have the same issue before on Discord.
It'll be fixed in 0.31, but unfortunately I have no solution for 0.30 other then not to use the potion on non player entities
